### PR TITLE
web: add LogPane component to HUD app

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: web/src/index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ let Main = () => {
           * New HUD work. HUD will eventually be promoted to the top-level component
           * and we'll delete App & friends
           */}
-        <Route exact path="/hud" component={HUD} />
+        <Route path="/hud" component={HUD} />
```
Comment (jazzdan) [inline web/src/index.js]: attn @nicks this is tricky. With exact matching on for this top level any path like `/hud/foo` or `/hud/foo/bar` would be routed to missing route. It seems to work as expected with exact matching off, I think this is due to the way we're doing this nested "switch" "statement"?

_grumble grumble react router_

File: web/src/LogPane.test.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import LogPane from './LogPane'
+import renderer from 'react-test-renderer';
+
+it('renders without crashing', () => {
+  let div = document.createElement('div')
+  Element.prototype.scrollIntoView = jest.fn()
```
Comment (nicks) [inline web/src/LogPane.test.tsx]: what's this for?

File: web/src/LogPane.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import React, { Component, createRef } from 'react';
+import Ansi from "ansi-to-react";
+
+type AnsiLineProps = {
+  line: string
+  key: string
```
Comment (nicks) [inline web/src/LogPane.tsx]: i'm surprised you need to specify key? all react child components have key, even if it's implicit

File: web/src/LogPane.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+
+  constructor(props: LogPaneProps) {
+    super(props)
+
+    this.state = {
+      autoscroll: true,
+    }
+
+    // TODO(dmiller): do I need to bind refreshAutoScroll?
```
Comment (nicks) [inline web/src/LogPane.tsx]: yes

in fact, i don't understand how this code can possibly work if you don't

File: web/src/index.js (hunk 1 above)
Comment (nicks) [reply web/src/index.js]: this is the behavior we want tho? /hud/foo should go to the missing route. what am i missing?

Comment (nicks) [reply web/src/index.js]: oh, nm, ya i understand, this is right

File: web/src/LogPane.test.tsx (hunk 2 above)
Comment (jazzdan) [reply web/src/LogPane.test.tsx]: `scrollIntoView` doesn't exist on a DOM elements in the testing environment for some reason? Without this the tests fail with a type error from node complaining that that function doesn't exist.

File: web/src/LogPane.tsx (hunk 3 above)
Comment (jazzdan) [reply web/src/LogPane.tsx]: Yep you're right.

**Comment by jazzdan** [comment]
There's a bug here where it looks like the logs are scrolling to the right instead of top down, will investigate.

**Comment by jazzdan** [comment]
In the spirit of prototype and with the knowledge that this is exactly the problem that I am bad at solving I'm going to merge this with one outstanding problem that I'd like help with.

Screenshot_2019-03-28 Tilt

The logs are showing up and streaming just fine afaict, but they're streaming vertically instead of horizontally. Any idea what might be causing this? My visual diff of the CSS rules that got applied to those divs seemed identical but the problem might lie elsewhere.

**Pull request merged**